### PR TITLE
Iohub dependency - [Testing PR]

### PR DESCRIPTION
**Pull request by Christianfoley**
The preprocessing components of the `iohub_dependency` branch are ready for testing as of most recent commit.

Some example code for running your configs:
```py
from micro_dl.cli.preprocess_script import pre_process
import micro_dl.utils.aux_utils as aux_utils

config_path = "your_config_path_here"
config = aux_utils.read_config(config_path)
pre_process(config)
```

I have had success with the following configs:
`/hpc/projects/CompMicro/projects/infected_cell_imaging/VirtualStaining/config_files/torch_gpLoading_2023_02/torch_config_nucl25D_iohubTest.yml`

The original and resulting preprocessed stores are at:
`/hpc/projects/CompMicro/projects/infected_cell_imaging/VirtualStaining/VirtualStain_NuclMem_A549_2023_02_07/Input_Nucl/A549_20X_iohubTest.zarr`
and:
`/hpc/projects/CompMicro/projects/infected_cell_imaging/VirtualStaining/VirtualStain_NuclMem_A549_2023_02_07/Input_Nucl/A549_20X_iohubTest_preprocessed.zarr`

Some attention will be required for training as some of the code with gunpowder still needs to be updated. Once it's ready we can test that in this thread as well.

### CODE REVIEW

File: micro_dl/evaluation/evaluation.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        self.preprocessing_config = self.torch_config["preprocessing"]
+
+        self.inference_metrics = {}
+        self.log_writer = SummaryWriter(log_dir=self.save_folder)
+    
+    def get_save_location(self):
+        """
+        Sets save location as specified in config files.
+        """
+        #TODO implement
```
Comment (ziw-liu) [inline micro_dl/evaluation/evaluation.py:30]: Maybe throw an error instead?

```suggestion
        #TODO implement
```
```py
raise NotImplementedError
```

File: micro_dl/inference/inference.py [hunk 2]
```diff
@@ -11,21 +11,21 @@
 
 import micro_dl.torch_unet.utils.model as model_utils
 import micro_dl.input.dataset as ds
-import microDL.micro_dl.utils.cli_utils as cli_utils
+import micro_dl.utils.cli_utils as cli_utils
 import micro_dl.utils.aux_utils as aux_utils
-import micro_dl.utils.normalize as normalize
 import micro_dl.utils.io_utils as io_utils
```
Comment (ziw-liu) [inline micro_dl/inference/inference.py:16]: Remove this line?

File: micro_dl/inference/inference.py [hunk 3]
```diff
@@ -316,364 +169,181 @@ def predict_large_image(self, input_image, model=None):
...
-            metrics_estimator.estimate_xyz_metrics(
-                target=target,
-                prediction=prediction,
-                pred_name=pred_name,
-            )
-            metrics_xyz = self._collapse_metrics_dict(
-                metrics_estimator.get_metrics_xyz().to_dict()
-            )
-            prediction_metrics["xyz"] = metrics_xyz
+        #Positions are unspecified and need to be read
```
Comment (Soorya19Pradeep) [inline micro_dl/inference/inference.py:219]: Currently the inference reads the positions specified in the inference section of the config yaml file and then proceeds to the following lines, where it also reads the data split yaml stored in the model directory specified. Should it be modified to chose either, based on decision 
- if we run it on a new dataset, take the position list mentioned in the config.
- if we run it on test set from the dataset used for generating the model, then use the data split file?

File: micro_dl/light/data.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    def _get_normalizer(self, source_channel: str, target_channel: str):
+        norm_meta = self.plate.zattrs["normalization"]
+        self.source_normalizer = NormalizeIntensity(
+            subtrahend=norm_meta[source_channel]["dataset_statistics"]["median"],
+            divisor=norm_meta[source_channel]["dataset_statistics"]["iqr"],
+        )
+        self.target_normalizer = NormalizeIntensity(
+            subtrahend=norm_meta[target_channel]["dataset_statistics"]["median"],
+            divisor=norm_meta[target_channel]["dataset_statistics"]["iqr"],
+        )
```
Comment (Christianfoley) [inline micro_dl/light/data.py:60]: This only allows access to dataset level normalization, fov-level normalization should also be possible.

**Pull request merged**